Addition of workflow to add issues and PRs to organization projects

mjprince opened the pull request:
<!-- Please replace #XX below with an existing issue number. Remove the line entirely if none exist. -->
Fixes #305 

## Description
Created a workflow to 
- automatically add new issues created in this repository to the CC Search backlog GitHub project.
- automatically add new PRs created in this repository to the CC Search active sprint GitHub project.

## Other information
<!-- Add any other information below or delete the "Other Information" line entirely. -->
I have attempted the bonus questions, but have not found a solution yet.

## Checklist
- [ ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `master` branch of the repository.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

<!-- Make sure you read and understand the following attestation. -->
## Developer Certificate of Origin
```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

mjprince committed:
Create main.yml

mjprince commented:
@kgodey Please could you have a look?

kgodey commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,28 @@
+      env:
+        GITHUB_TOKEN: ${{ secrets.repositoryToken }}
+        GITHUB_PROJECT_URL: https://github.com/orgs/creativecommons/projects/7
+        GITHUB_PROJECT_COLUMN_NAME: Ticket Work Required
```
This should be the "In Progress" column.

mjprince replied:
I have corrected this in my recent update.
![pr](https://user-images.githubusercontent.com/45537681/76521207-a1a9fa00-6464-11ea-8bbe-de1dd6f711e0.png)

kgodey commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,28 @@
+      uses: docker://takanabe/github-actions-automate-projects:v0.0.1
+      if: github.event_name == 'issues' && github.event.action == 'opened'
+      env:
+        GITHUB_TOKEN: ${{ secrets.repositoryToken }}
```
please set this variable to `secrets.REPO_GITHUB_TOKEN`

mjprince replied:
I have corrected this in my recent update.
![pr](https://user-images.githubusercontent.com/45537681/76521224-aa9acb80-6464-11ea-8d5b-52fe3c2166cc.png)

mjprince committed:
Update main.yml

Add pull requests to the "In progress" column of the Active sprint project.
Provide the correct github token to add issues and pull requests to the corresponding projects.

kgodey reviewed (changes requested):
@mjprince This looks like it will work, good job! I think it would be cleaner to do it in one job, though, without repeating code. Use the workflow file in `cccatalog-api`: as an example: https://github.com/creativecommons/cccatalog-api/blob/master/.github/workflows/IssueAndPR.yml

Also, I renamed the github token to `ORG_GITHUB_TOKEN`, please use that for consistency.

mjprince committed:
Update main.yml

Merge two jobs into one.

mjprince commented:
@kgodey I have done that. Please have a look.

kgodey commented:
Thanks @mjprince!
